tpm2/pcr_profile.go

```diff
@@ -728,6 +735,279 @@ func (p *PCRProtectionProfile) String() string {
 	return s.String() + "\n"
 }
 
+var (
+	savedPCRBits           = bits.Len16(uint16(maxPCR))
+	savedDigestMask uint32 = (1 << (32 - savedPCRBits)) - 1
+	savedPCRMask    uint16 = (1 << savedPCRBits) - 1
+	maxSavedDigests int    = int(savedDigestMask)
+)
+
+// savedPCRProtectionProfilePCRAndDigest provides a way of storing a PCR
+// index and digest index as a uint32 in a saved PCR profile, with the
+// lower 11 bits used to store the PCR index and the upper 22 bits used
+// to store a digest index, which references a digest in a list.
+type savedPCRProtectionProfilePCRAndDigest uint32
+
+func (i savedPCRProtectionProfilePCRAndDigest) PCR() uint16 {
+	return uint16(i) & savedPCRMask
+}
+
+func (i savedPCRProtectionProfilePCRAndDigest) Digest() uint32 {
+	return (uint32(i) >> savedPCRBits) & savedDigestMask
+}
+
+func newSavedPCRProtectionProfilePCRAndDigest(pcr uint16, digest uint32) savedPCRProtectionProfilePCRAndDigest {
+	return savedPCRProtectionProfilePCRAndDigest(uint32(pcr&savedPCRMask) | ((digest & savedDigestMask) << savedPCRBits))
+}
+
+type savedPCRProtectionProfilePCREventInstrData struct {
+	Alg          tpm2.HashAlgorithmId
+	PCRAndDigest savedPCRProtectionProfilePCRAndDigest
+}
+
+type savedPCRProtectionProfileAddPCRValueFromTPMInstrData struct {
+	Alg tpm2.HashAlgorithmId
+	PCR uint16
+}
+
+// savedPCRProtectionProfileInstrData represents the data associated with a
+// single instruction in a saved PCR profile.
+type savedPCRProtectionProfileInstrData struct {
+	AddPCRValue        *savedPCRProtectionProfilePCREventInstrData
+	AddPCRValueFromTPM *savedPCRProtectionProfileAddPCRValueFromTPMInstrData
+	ExtendPCR          *savedPCRProtectionProfilePCREventInstrData
+}
+
+func (d *savedPCRProtectionProfileInstrData) Select(selector reflect.Value) interface{} {
```

Comment:
this should have a comment like:

// Select implements the mu.Union interface.